### PR TITLE
Feature Creation copy for API doc index.rst

### PR DESCRIPTION
**Pull request by Morgan-Sell**
Wrote copy for creation module API Docs index.rst. Copy achieved A++ content grade.

### CODE REVIEW

File: docs/api_doc/creation/index.rst [hunk 1]
```diff
@@ -3,15 +3,118 @@
...
-Feature-engine's creation transformers create and add new features to the dataframe
-by either combining or transforming existing features.
+Feature creation is a  feature engineering technique that consists of 
+constructing new variables from the dataset’s original features. By combining 
+or amalgamating two or more variables, a new variable is developed to improve 
+the performance of a machine learning model.
 
-.. toctree::
-   :maxdepth: 1
+One of the most common feature-creation methods in data science is one-hot
```
Comment (solegalli) [inline docs/api_doc/creation/index.rst:11]: Can we link one-hot encoding to this article please: https://www.blog.trainindata.com/one-hot-encoding-categorical-variables/

File: docs/api_doc/creation/index.rst [hunk 2]
```diff
@@ -3,15 +3,118 @@
...
+including evaluating the relationship between the independent or predictor 
+variable and the dependent/target variable. It is useful to have domain knowledge 
+of the dataset when performing feature creation, making it one of the more 
+creative approaches to improve a predictive model’s performance. 
+
+Lastly, a data scientist should be mindful that creating features increases the 
+number of features, which increases the dimensionality. This may have downstream 
+effects depending on the machine learning algorithm being used. 
+
+Creating New Features with feature-engine
```
Comment (solegalli) [inline docs/api_doc/creation/index.rst:25]:
```suggestion
Creating New Features with Feature-engine
```

File: docs/api_doc/creation/index.rst [hunk 3]
```diff
@@ -3,15 +3,118 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+feature-engine’s creation transformers generate new features that are added to the 
+pandas dataframe by either combining or transforming existing features. The 
+mathematical or statistical operations used in the Python package vary depending 
+on the selected transformer. The transformers are to be used with numerical 
+variables, e.g., integer and float data types.
+
+Summary of feature-engine’s feature-creation transformers:
```
Comment (solegalli) [inline docs/api_doc/creation/index.rst:34]:
```suggestion
Summary of Feature-engine’s feature-creation transformers:
```

File: docs/api_doc/creation/index.rst [hunk 4]
```diff
@@ -3,15 +3,118 @@
...
+``ColumnTrasnformer()`` but differs in implementation.
+
+One useful application is to use ``SklearnTransformerWrapper()`` with Scikit-learn’s 
+``PolynomialFeatures()``. This is appropriate in feature engineering when an independent 
+variable does not have a linear or monotonic relationship with the target variable.
+
+A data scientist may perform feature creation when they believe that the interaction 
+between two variables can have a strong relationship with the target variable.
+
+feature-engine in Practice
```
Comment (solegalli) [inline docs/api_doc/creation/index.rst:58]:
```suggestion
Feature-engine in Practice
```

File: docs/api_doc/creation/index.rst [hunk 5]
```diff
@@ -3,15 +3,118 @@
...
+   !unzip -q wine-quality-dataset.zip
+   
+   # read raw data and extract relevant features
+   data = pd.read_csv("WineQT.csv")
+   df = data[["alcohol", "pH", "residual sugar"]].copy()
+   
+   # perform feature transformation
+   rf = RelativeFeatures(variables=["alcohol", "pH"], reference=["residual sugar"], func=["div"])
+   rf.fit(df)
+   df_tr = rf.transform(df)
```
Comment (solegalli) [inline docs/api_doc/creation/index.rst:92]: can we break this massive code block into smaller chunks and show the intermediate results in between?

For example load data and show the first rows, and then also show the result of the relative features transformation?

File: docs/api_doc/creation/index.rst [hunk 6]
```diff
@@ -3,15 +3,118 @@
...
+scoring metrics. With feature creation, new data points are constructed that improve a machine 
+learning algorithm’s learning process. 
+
+To ensure the creation of high-quality features, thorough data mining is strongly encouraged to 
+understand the relationships among the various independent variables.
+
+One must be mindful of excessive feature creation, in addition to high dimensionality, unrestrained 
+feature creation can cause overfitting, i.e., high variance.  Predictive models with high variance 
+are susceptible to outliers. That is why we  recommend using feature creation in conjunction with 
+feature selection, which feature-engine has a module for.
```
Comment (solegalli) [inline docs/api_doc/creation/index.rst:113]:
```suggestion
feature selection, which Feature-engine has a module for.
```

**Review by solegalli — COMMENTED** [review]
Morgan this looks just great! Thank you so much! 

I made a few suggestions (mostly capitalize Feature-engine). You can just accept them here and then pull the changes to your branch. Or can update your branch and push, as you wish.

The errors in the tests are due to pandas and sklearn new releases. But I fixed that yesterday. So if you pull the latest main version onto your main branch, then check out your seo branch and rebase main onto it, the tests should pass

This is almost ready to go! Thank you!

**Comment by Morgan-Sell** [comment]
hola @solegalli, me alegro que te gustelo! I had to get creative, e.g., "real-world experimentation" ;)

Would you like for all mentions of "Feature-engine" to be capitalized? I ask because you skipped over a few. Lmk and I'll revise the copy. 

Voy a comenzar escribir el texto de time series. Vamooossss ;)

**Comment by solegalli** [comment]
> Would you like for all mentions of "Feature-engine" to be capitalized?

Yes please! Thank you!

**Comment by Morgan-Sell** [comment]
done!

**Comment by solegalli** [comment]
Could you please rebase main onto the feature branch to fix the tests?

**Comment by Morgan-Sell** [comment]
hola @solegalli, I added a toctree Sphinx at the bottom of the .rst file. Is that correct? Once we're good on the toctree, we're good to go!

**Comment by solegalli** [comment]
Hey @Morgan-Sell 

Something went wrong with rebase. I should see here only the files that you changed for this PR. But I see 29 files instead.

Could you please check?

**Comment by Morgan-Sell** [comment]
@solegalli, I see what you're saying. Fortunately, I didn't perform major changes after rebasing. Is the best approach the following?

1. Checkout commit # `7b422bb78da3279e7c9ee3f6c613d57ea6de7685` which is the commit prior
2. Rebase correctly
3. Perform the last changes
4. Add/commit/push to this PR

**Comment by solegalli** [comment]
it sounds ok to me

**Comment by Morgan-Sell** [comment]
Hi @solegalli, I tried the above approach and I ended up with the same result.

I followed the directions on feature-engine contribute docs, so I'm unsure what happened. However, will this be an issue when merging to feature-engine main? It seems somehow I merged `main` onto `seo_copy_creation` and didn't squah the commits.

I know it isn't ideal, but I believe/hope it will be fine in the end.

**Pull request closed without merging**